guides/source/active_model_basics.md

```diff
@@ -44,76 +67,331 @@ class EmailContact
+
+irb> person.name = "Jane"
+irb> person.name
+=> "Jane"
+
+# casts the string to a date set by the attribute
+irb> person.date_of_birth = "2020-01-01"
+irb> person.date_of_birth
+=> Wed, 01 Jan 2020
+irb> person.date_of_birth.class
+=> Date
+
+# uses the default value set by the attribute
+irb> person.active
 => true
-irb> email_contact.persisted?
+
+# casts the string to a boolean set by the attribute
+irb> person.active = "0"
+irb> person.active
 => false
 ```
 
-Any class that includes `ActiveModel::API` can be used with `form_with`,
-`render` and any other Action View helper methods, just like Active Record
-objects.
+Some additional methods described below are available when using
+`ActiveModel::Attributes`.
+
+#### Method: `attribute_names`
+
+The `attribute_names` method returns an array of attribute names.
+
+```irb
+irb> Person.attribute_names
+=> ["name", "date_of_birth", "active"]
+```
+
+#### Method: `attributes`
+
+The `attributes` method returns a hash of all the attributes with their names as
+keys and the values of the attributes as values.
+
+```irb
+irb> person = Person.new
+irb> person.name = "John"
+irb> person.date_of_birth = "1998-01-01"
+irb> person.active = false
+
+irb> person.attributes
+=> {"name"=>"John", "date_of_birth"=>Thu, 01 Jan 1998, "active"=>false}
+```
+
+### Attribute Assignment
+
+`ActiveModel::AttributeAssignment` allows you to set an object's attributes by
+passing in a hash of attributes with keys matching the attribute names. This is
+useful when you want to set multiple attributes at once.
+
+Consider the following class:
+
+```ruby
+class Person
+  include ActiveModel::AttributeAssignment
+
+  attr_accessor :name, :date_of_birth, :active
+end
+```
+
+You can set multiple attributes at once using the `assign_attributes` method:
+
+```irb
+irb> person = Person.new
+
+# using the `assign_attributes` method to set multiple attributes at once
+irb> person.assign_attributes(name: "John", date_of_birth: "1998-01-01", active: false)
+
+# check the values of the attributes that were assigned
+irb> person.name
+=> "John"
+irb> person.date_of_birth
+=> Thu, 01 Jan 1998
+irb> person.active
+=> false
+```
+
+If the passed hash responds to the `permitted?` method and the return value of
+this method is `false`, an `ActiveModel::ForbiddenAttributesError` exception is
+raised.
+
+#### Method alias: `attributes=`
+
+The `assign_attributes` method has an alias `attributes=`.
+
+INFO: A method alias is a method that performs the same action as another
+method, but is called something different.
+
+The following example demonstrates the use of the `attributes=` method to set
+multiple attributes at once:
+
+```irb
+irb> person = Person.new
+
+# using the `attributes= method` to set multiple attributes at once
+irb> person.attributes = { name: "John", date_of_birth: "1998-01-01", active: false }
+
+irb> person.name
+=> "John"
+
+# using the `assign_attributes` method to set multiple attributes at once
```

Comment:
This particular case it isn't multiple attributes. I'm also up for removing this comment as well, as it may not be adding much.
```suggestion
# using the `assign_attributes` method
```